packages/frontend/src/analyticsEvents.ts

```diff
@@ -0,0 +1,77 @@
+/**
+ * Analytics event name constants for Application Insights tracking
+ *
+ * Event naming convention: [Category].[Action].[Status]
+ * - Category: The main area/component (GUI, Dialog, User, SavedSearch)
+ * - Action: The specific action being performed (Click, Move, Language, etc.)
+ * - Status: The outcome (Success, Failed) - no "Attempt" events
+ */
+
+// GUI Action Events
+export const ANALYTICS_EVENTS = {
+  // GUI Actions
+  GUI_ACTION_CLICK: 'GUI.Action.Click',
+  GUI_ACTION_CANCELLED: 'GUI.Action.Cancelled',
+  GUI_ACTION_SUCCESS: 'GUI.Action.Success',
+  GUI_ACTION_FAILED: 'GUI.Action.Failed',
+
+  // Dialog Transmissions
+  DIALOG_TRANSMISSIONS_EXPAND: 'Dialog.Transmissions.Expand',
+  DIALOG_TRANSMISSIONS_COLLAPSE: 'Dialog.Transmissions.Collapse',
+
+  // User Actions
+  USER_LOGOUT: 'User.Logout',
+  USER_LANGUAGE_CHANGE_SUCCESS: 'User.Language.ChangeSuccess',
+  USER_LANGUAGE_CHANGE_FAILED: 'User.Language.ChangeFailed',
+
+  // Dialog Move Actions
+  DIALOG_MOVE_TO_INBOX_SUCCESS: 'Dialog.Move.ToInbox.Success',
+  DIALOG_MOVE_TO_INBOX_FAILED: 'Dialog.Move.ToInbox.Failed',
+  DIALOG_MOVE_TO_ARCHIVE_SUCCESS: 'Dialog.Move.ToArchive.Success',
+  DIALOG_MOVE_TO_ARCHIVE_FAILED: 'Dialog.Move.ToArchive.Failed',
+  DIALOG_MOVE_TO_BIN_SUCCESS: 'Dialog.Move.ToBin.Success',
+  DIALOG_MOVE_TO_BIN_FAILED: 'Dialog.Move.ToBin.Failed',
+
+  // Saved Search Actions
+  SAVED_SEARCH_CREATE_SUCCESS: 'SavedSearch.Create.Success',
+  SAVED_SEARCH_CREATE_FAILED: 'SavedSearch.Create.Failed',
+  SAVED_SEARCH_DELETE_SUCCESS: 'SavedSearch.Delete.Success',
+  SAVED_SEARCH_DELETE_FAILED: 'SavedSearch.Delete.Failed',
+  SAVED_SEARCH_UPDATE_SUCCESS: 'SavedSearch.Update.Success',
```

Comment:
This name is probably a bit misleading elsewhere in the code, but the only thing the user can do here, as might be aware of, is give the saved search a name. And of course, it's possible to update without giving a name. Basically doing nothing. 

A proposal:

What if we have an event called `SAVED_SEARCH_NAME_UPDATED` which requires a success update of a saved search AND the payload had a title. 
I think this is fruitful to see whether the users understand the mechanics here.